brats/transformations.py

```diff
@@ -73,3 +79,120 @@ def hist_match(source: np.ndarray, template: np.ndarray) -> np.ndarray:
                                     template_values)
         result[positive] = interp_t_values[bin_idx]
         return result
+
+
+@singledispatch
+def reorder(img: typing.Any):
+    return TypeError("Img should be either np.ndarray of torch.Tensor")
+
+
+@reorder.register(np.ndarray)
+def _(img: np.ndarray) -> np.ndarray:
+    transformed = np.moveaxis(img, [0, 1, 2, 3], [2, 3, 1, 0])
+    return transformed
+
+
+@reorder.register(torch.Tensor)
+def _(img: torch.Tensor) -> torch.Tensor:
+    transformed = img.permute(3, 2, 0, 1)
+    return transformed
+
+
+class NiftiToTorchDimensionsReorderTransformation:
+    """
+    Changes dimensions order from nifti (H,W,D,C) to torch convention (C,D,H,W).
+    """
+
+    def __call__(self, img: typing.Union[np.ndarray, torch.Tensor]) -> typing.Union[np.ndarray, torch.Tensor]:
+        assert len(img.shape) == 4, "Tensor should have 4 dimensions (H,W,D,C)"
+
+        transformed = reorder(img)
+        return transformed
+
+
+@singledispatch
+def add_channel_dim(img: typing.Any):
+    return TypeError("Img should be either np.ndarray of torch.Tensor")
+
+
+@add_channel_dim.register(np.ndarray)
+def _(img: np.ndarray) -> np.ndarray:
+    transformed = np.expand_dims(img, 3)
+    return transformed
+
+
+@add_channel_dim.register(torch.Tensor)
+def _(img: torch.Tensor) -> torch.Tensor:
+    transformed = torch.unsqueeze(img, 0)
+    return transformed
+
+
+class AddChannelDimToMaskTransformation:
+    """
+    Adds additional channel dimension
+    """
+
+    def __call__(self, img: typing.Union[np.ndarray, torch.Tensor]) -> typing.Union[np.ndarray, torch.Tensor]:
+        assert img.ndim == 3, "Img should have 3 dimensions (D,H,W)"
+        transformed = add_channel_dim(img)
+        return transformed
+
+
+@singledispatch
+def binarize(img: typing.Any):
+    return TypeError("Img should be either np.ndarray of torch.Tensor")
+
+
+@binarize.register(np.ndarray)
+def _(img: np.ndarray) -> np.ndarray:
+    transformed = img.copy()
```

Comment:
I think `np.copy(img)` would be safer. `np.ndarray.copy(img)` always converts to `C` memory layout if possible `np.copy()` just chooses one that is most similar.

Reply:
Or you can just pass `order=K` xd